PsfLauncher/main.cpp

```diff
@@ -71,185 +64,199 @@ int launcher_main(PWSTR args, int cmdShow) noexcept try
         throw_win32(ERROR_NOT_FOUND, "Error: could not find matching appid in config.json and appx manifest");
     }
 
-    auto exeName = appConfig->get("executable").as_string().wide();
     auto dirPtr = appConfig->try_get("workingDirectory");
-    auto dirStr = dirPtr ? dirPtr->as_string().wide() : nullptr;
-
+    auto dirStr = dirPtr ? dirPtr->as_string().wide() : L"";
     auto exeArgs = appConfig->try_get("arguments");
-    auto exeArgString = exeArgs ? exeArgs->as_string().wide() : (wchar_t*)L"";
-    auto monitor = PSFQueryAppMonitorConfig();
-
 
     // At least for now, configured launch paths are relative to the package root
     std::filesystem::path packageRoot = PSFQueryPackageRootPath();
-    auto exePath = packageRoot / exeName;
 
-    // Allow arguments to be specified in config.json now also.
-    std::wstring cmdLine = L"\"" + exePath.filename().native() + L"\" " + exeArgString + L" " + args;
+    ErrorInformation error;
+    bool isPowershellInstalled = false;
+    error = CheckIfPowershellIsInstalled(isPowershellInstalled);

```

Comment:
define & init in one line

Reply:
I don't understand, is something wrong with defining and initializing a wstring on the same line?